src/toil/job.py

```diff
@@ -667,18 +667,20 @@ def _cleanLocalTempDir(self, cacheSize):
             """
             #Remove files so that the total cached files are smaller than a cacheSize
             
-            #List of pairs of (fileSize, fileStoreID) for cached files
-            cachedFileSizes = map(lambda x : (os.stat(self.jobStoreFileIDToCacheLocation[x]).st_size, x), 
```

Comment:
@benedictpaten seems to like map + lambda but often a comprehension is more concise.

AFAICT, a lot of this can be turned to iterators and generators instead of lists.

No show stoppers, though.